File: Java-Projects/OOP-Geometry/src/Circle.java
public class Circle extends PlaneShape {
    private double radius;

    public Circle(Vertex center, double radius) {
        super(center);
        this.setRadius(radius);
    }

    public double getRadius() {
        return this.radius;
    }

    public void setRadius(double radius) {
        this.radius = Math.abs(radius);
    }

    public double getArea() {
        return Math.PI * this.radius * this.radius;
    }

    public double getPerimeter() {
        return Math.PI * this.radius * 2;
    }

    @Override
    public String toString() {
        return super.toString() + String.format("\nCenter: %s\nRadius: %s",
                this.vertices[0], this.getRadius());
    }
}

File: C-Sharp-Projects/C-Sharp-Advanced/Encapsulation-And-Polymorphism/01-Shapes/Circle.cs
using System;

namespace _01_Shapes
{
    public class Circle : IShape
    {
        public Circle(double radius)
        {
            this.Radius = radius;
        }

        public double Radius { get; set; }

        public double CalculateArea()
        {
            return Math.PI * this.Radius * this.Radius;
        }

        public double CalculatePerimeter()
        {
            return Math.PI * 2 * this.Radius;
        }
    }
}
